src/VisualStudio/Core/Def/Utilities/ClipboardHelpers.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace Microsoft.VisualStudio.LanguageServices.Utilities
+{
+    internal static class ClipboardHelpers
+    {
+        [DllImport("user32.dll")]
+        [PreserveSig]
+        private static extern int GetPriorityClipboardFormat([In] uint[] formatPriorityList, [In] int formatCount);
+
+        /// <summary>
+        /// Determines if clipboard data exists for the text format without blocking
+        /// on <see cref="Clipboard.GetText()"/> or <see cref="Clipboard.GetDataObject()"/> for cases where the clipboard is being
+        /// delay rendered. See https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1463413/ 
+        /// </summary>
+        public static bool CanGetText()
+        {
+            const uint TextFormatId = 1;          // ID for clipboard text format (CF_TEXT)
```

Comment:
MSDN says this is the format to use for ANSI text, there is also a CF_UNICODETEXT, do we need to accept unicode text that can't be expressed via CF_TEXT? Or is the assumption everything is UTF8 encoded and thus will always be accessible as if it were ANSI text?

Reply:
I think you're looking at an older commit? This was changed to use IDataObject